src/Git/GitStatusListCtrl.cpp

```diff
@@ -2764,7 +2763,22 @@ void CGitStatusListCtrl::StartDiffWC(int fileindex)
 	CTGitPath file1 = *ptr;
 	file1.m_Action = 0; // reset action, so that diff is not started as added/deleted file; see issue #1757
 
-	CGitDiff::Diff(GetParentHWND(), &file1, &file1, GIT_REV_ZERO, m_CurrentVersion.ToString(), false, false, 0, !!(GetAsyncKeyState(VK_SHIFT) & 0x8000));
+	CString version;
+	if (parent)

```

Comment:
Wouldn't it be easier to use `m_currentVersion.ToString() + L"~1"`?

Reply:
"~1" allows to get parent commit for the first parent, and existing code allows to retrieve also other parents.
Code for evaluating parent commit was already implemented in "Revert to parent revision" action, I only did some refactoring to reuse it.